src/reconcile/rec_visibility.c

```diff
@@ -1519,10 +1541,10 @@ __wti_rec_upd_select(WT_SESSION_IMPL *session, WTI_RECONCILE *r, WT_INSERT *ins,
      * functions perform the history store truncation for this key.
      */
     if (!WT_IS_HS(session->dhandle) && upd_select->tombstone != NULL &&
-      !F_ISSET(upd_select->tombstone, WT_UPDATE_RESTORED_FROM_DS | WT_UPDATE_RESTORED_FROM_HS) &&
+      !F_ISSET(upd_select->tombstone,
+        WT_UPDATE_RESTORED_FROM_DS | WT_UPDATE_RESTORED_FROM_HS | WT_UPDATE_RESTORED_FROM_DELTA) &&
```

Comment:
Is the addition of the flag `WT_UPDATE_RESTORED_FROM_DELTA` related to the changes here?
If yes, we would benefit from a new variable to make the condition easier to read.